libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1565,6 +1565,16 @@ void TypeChecker::endVisit(ElementaryTypeNameExpression const& _expr)
 
 void TypeChecker::endVisit(Literal const& _literal)
 {
+	if (_literal.looksLikeAddress())
+	{
+		if (_literal.passesAddressChecksum())
```

Comment:
I did not do it that way this time, because I think it is nice that you cannot do
`uint x = 0x<correctly checksummed address>` now.